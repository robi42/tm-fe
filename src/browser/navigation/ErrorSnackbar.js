// @flow
import React from 'react';
import Snackbar from 'react-toolbox/lib/snackbar';
import { connect } from 'react-redux';
import { Durations } from '../../common/lib/utils';
import { appError } from '../../common/app/actions';

type State = {
  active: boolean,
};
type Props = {
  error: ?string,
  appError: Function,
};

const {
  ONE_SECOND_AS_MILLIS: ONE_SECOND,
  THREE_SECONDS_AS_MILLIS: THREE_SECONDS,
} = Durations;

export class ErrorSnackbar extends React.Component {

  state: State = {
    active: this.props.error != null,
  };

  componentWillReceiveProps(nextProps: Props) {
    if (nextProps.error != null) {
      this.setState({ active: true });
    }
  }

  props: Props;

  handleClick() {
    this.handleTimeout();
  }

  handleTimeout() {
    this.setState({ active: false });
    setTimeout(() => {
      this.props.appError({ message: null });
    }, ONE_SECOND);
  }

  render() {
    return (
      <Snackbar
        action="Dismiss"
        active={this.state.active}
        label={this.props.error}
        timeout={THREE_SECONDS}
        onClick={() => this.handleClick()}
        onTimeout={() => this.handleTimeout()}
        type="warning"
      />
    );
  }

}

export default connect(state => ({
  error: state.app.error,
}), { appError })(ErrorSnackbar);
