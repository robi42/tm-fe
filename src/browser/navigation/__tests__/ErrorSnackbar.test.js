import React from 'react';
import renderer from 'react-test-renderer';
import { ErrorSnackbar } from '../ErrorSnackbar';

test('<ErrorSnackbar /> renders', () => {
  const tree = renderer.create(
    <ErrorSnackbar error="Something went wrong" />,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
