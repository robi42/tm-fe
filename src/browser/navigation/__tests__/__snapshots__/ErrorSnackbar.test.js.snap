exports[`test <ErrorSnackbar /> renders 1`] = `null`;
